src/core_landice/mode_forward/mpas_li_thermal.F

```diff
@@ -2771,18 +2996,27 @@ subroutine basal_melt_grounded_ice(&
             !       and overlying layers with a different enthalpy also melt.
 
             netBasalFlux = basalFrictionFlux(iCell) + basalHeatFlux(iCell) + basalConductiveFlux(iCell)  ! W/m^2
+            !call mpas_log_write("netBasalFlux G condFlux $r $r $r", realArgs=(/netBasalFlux, basalHeatFlux(iCell),basalConductiveFlux(iCell)/))
 
             if (abs(netBasalFlux) < eps11) then  ! netBasalFlux might be slightly different from zero
                                                  ! because of rounding errors; if so, then zero out
                netBasalFlux = 0.0_RKIND
             endif
 
             if (trim(config_thermal_solver) == 'enthalpy') then
-               groundedBasalMassBal(iCell) = -netBasalFlux / (latent_heat_ice * rhoi - enthalpy(nVertLevels,iCell))
+               !groundedBasalMassBal(iCell) = -netBasalFlux / (latent_heat_ice * rhoi - enthalpy(nVertLevels,iCell))
+               groundedBasalMassBal(iCell) = -netBasalFlux / (latent_heat_ice * rhoi)
+               ! TZ: Don't understand yet why WHL added enthalpy(nVertLevels,iCell) here
+                basalWaterThickness(iCell) = basalWaterThickness(iCell) - deltat*groundedBasalMassBal(iCell)
+                ! TZ: allow basalWaterThickness freely accumulate here. Change it to something else for difference cases
```

Comment:
The free accumulation here is just for the benchmark experiment. I don't know if it should link to the subglacial hydrology model so that the water can drain in a certain way?

Reply:
We should not worry about trying to couple it to the subglacial hydrology model for this PR.  But we should make sure there is some mechanism for preventing unrealistic build-up of water.  In the Hewitt and Schoof paper they say:

> A cap on porosity and removal of additional water are often implemented by adding a term −ρwLD(φ) to the right-hand side of Eq. (9), where D(φ) is a “drainage function” with units s−1 (e.g. Greve, 1997; Aschwanden et al., 2012).

We should try to include something like that if it's not already in there and it's not too hard.

Reply:
Ok. I'll work on that and let you review it.

Reply:
After looking around more, it looks like we already have a max water fraction function here:
https://github.com/MPAS-Dev/MPAS-Model/blob/landice/develop/src/core_landice/mode_forward/mpas_li_thermal.F#L2788-L2816
That is fine for now, so you should not add anything more complicated now.

Reply:
Ok. Sounds good to me.